zingo-sync/src/sync.rs

```diff
@@ -67,8 +102,33 @@ where
     );
     scan_ranges.push(chain_tip_scan_range);
 
+    if scan_ranges.is_empty() {
+        panic!("scan ranges should never be empty after updating")
+    }
+
     // TODO: add logic to combine chain tip scan range with wallet tip scan range
     // TODO: add scan priority logic
 
     Ok(())
 }
+
+// returns `None` if there are no more ranges to scan
+fn prepare_next_scan_range(sync_state: &SyncState) -> Option<ScanRange> {
```

Comment:
good unless this duplicates something already present in LRZ

Reply:
no it seems LRZ have not implemented any logic for selecting/manipulating scan ranges. its left to the consumer in `suggest_scan_ranges` currently